src/routes/homepage.tsx

```diff
@@ -0,0 +1,36 @@
+import { Container, Stack, Typography } from "@mui/material";
+import { auth, database } from "../firebase";
+import { get, ref } from "firebase/database";
+import { useEffect, useState } from "react";
+
+export default function Homepage() {
+  const [firstName, setFirstName] = useState("");
+  useEffect(() => {
+    get(ref(database, "users/" + auth.currentUser?.uid + "/firstName"))
+      .then((snapshot) => {
+        if (snapshot.exists()) {
+          setFirstName(snapshot.val());
+          console.log(firstName);
+        } else {
+          console.log("No data available");
+        }
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  });
```

Comment:
need to add useEffect dependancy array.
I think the IDE should tell you about missing dependancies>
Maybe need to configure the esline rules to get triggered for this.